transports/email/src/main/java/org/mule/transport/email/RetrieveMessageReceiver.java

```diff
@@ -101,32 +95,34 @@ protected void doConnect() throws Exception
             @Override
             public void doWork(RetryContext context) throws Exception
             {
-                logger.info("Attemping to establish connection with message retrieval server");
-                SessionDetails session = RetrieveMessageReceiver.this.castConnector().getSessionDetails(endpoint);
-
-                Store store = session.newStore();
-                store.connect();
-                RetrieveMessageReceiver.this.folder = store.getFolder(castConnector().getMailboxFolder());
-                if (castConnector().getMoveToFolder() != null)
+                synchronized (connectionLock)
```

Comment:
Add comments explaining the need for the lock

Reply:
Comment added